File: server/src/main/com/broll/gainea/server/core/goals/impl/all/G_Survive.java
package com.broll.gainea.server.core.goals.impl.all;

import com.broll.gainea.server.core.battle.BattleResult;
import com.broll.gainea.server.core.bot.strategy.GoalStrategy;
import com.broll.gainea.server.core.goals.GoalDifficulty;
import com.broll.gainea.server.core.goals.RoundGoal;
import com.broll.gainea.server.core.objects.BattleObject;

public class G_Survive extends RoundGoal {
    private final static int TARGET = 5;

    public G_Survive() {
        super(GoalDifficulty.EASY, "Verliere für " + TARGET + " Runden keine Einheit", TARGET);
    }

    @Override
    public void killed(BattleObject unit, BattleResult throughBattle) {
        if (unit.getOwner() == player) {
            resetRounds();
        }
    }

    @Override
    public void check() {
        progressRound();
    }

    @Override
    public void botStrategy(GoalStrategy strategy) {
        //todo
    }
}

File: server/src/main/com/broll/gainea/server/core/goals/impl/all/G_Survive.kt
package com.broll.gainea.server.core.goals.impl.all

import com.broll.gainea.server.core.battle.BattleResult
import com.broll.gainea.server.core.bot.strategy.GoalStrategy
import com.broll.gainea.server.core.goals.GoalDifficulty
import com.broll.gainea.server.core.goals.RoundGoal
import com.broll.gainea.server.core.objects.Unit

class G_Survive :
    RoundGoal(GoalDifficulty.EASY, "Verliere für " + TARGET + " Runden keine Einheit", TARGET) {
    override fun unitKilled(unit: Unit, throughBattle: BattleResult?) {
        if (unit.owner === player) {
            resetRounds()
        }
    }

    override fun check() {
        progressRound()
    }

    override fun botStrategy(strategy: GoalStrategy) {
        //todo
    }

    companion object {
        private const val TARGET = 5
    }
}
